packages/jupyter-ai-magics/jupyter_ai_magics/magics.py

```diff
@@ -177,22 +170,31 @@ def _ai_env_status_for_provider_markdown(self, provider_id):
         ):
             return na_message  # No emoji
 
+        # Avoid composing strings, to make localization easier in the future
+        not_set_title = "You have not set this environment variable, so you cannot use this provider's models."
+        set_title = "You have set this environment variable, so you can use this provider's models."
+        env_status_ok = False
```

Comment:
> Avoid composing strings, to make localization easier in the future

1. I strongly advise against this pattern. We shouldn't be imposing restrictions on our coding style in the off-chance that we are able to crowdsource/fund a team of translators to go through our entire application and localize every string. Also, given that this hasn't been discussed with the team, the rest of the codebase likely doesn't live up to the standard imposed here.

2. For statically-defined (i.e. hard-coded) constants like these, please follow convention by using all-caps and moving them to the top-level scope, e.g. `NOT_SET_TITLE = ...`. You'll probably need 2 more constant strings for the `multienv` case.

Reply:
#152 concerns future localization (l10n) of strings, and although it hasn't been prioritized yet, setting strings in a l10n-friendly way will make it easier for developers (whether on our team or not) to localize them in the future. I can move them to constants.

Reply:
> off-chance that we are able to crowdsource/fund a team of translators to go through our entire application and localize every string

For what it's worth, most of the Jupyter stack is translated by community, with perfect coverage for languages like Chinese or French. This extension is rather an outlier.

Reply:
> setting strings in a l10n-friendly way will make it easier for developers (whether on our team or not) to localize them in the future. I can move them to constants.

I'm lost here. How do my suggestions make it more difficult for devs to localize these in the future? Either way, the value is statically-defined and stored in a variable.